### PR TITLE
[TAPP-194] Refactor Application Form into Component and View

### PR DESCRIPTION
**Pull request by thezane**
Please see issue #194 for more details.

### CODE REVIEW

File: frontend/src/components/application-form.js [hunk 1]
```diff
@@ -1,12 +1,59 @@
...
-    type: a string that determines whether the input is a default input field, a textbox, or a binary radio choice
-    field_name: the key name for the field
-    validate: possible validations, not yet implemented
-    required: whether or not the field is required to be filled
-*/
+import React from "react";
+
+export const DefaultInput = ({ label, curValue, onChange, required }) => (
+    <div>
+        <label style={{ width: "20vw" }}>
```
Comment (siefkenj) [inline frontend/src/components/application-form.js:5]: Inline styles should be avoided. Use a `className` on the surrounding `div`

File: frontend/src/components/application-form.js [hunk 2]
```diff
@@ -1,12 +1,59 @@
...
+                checked={curValue === "true"}
+                onChange={onChange}
+            />
+            Yes
+        </label>
+        <label>
+            <input
+                type="radio"
+                value="false"
+                checked={curValue === "false"}
```
Comment (siefkenj) [inline frontend/src/components/application-form.js:49]: What? `curValue` is required to be a string? That seems weird...

File: frontend/src/components/application-form.js (hunk 1 above)
Comment (siefkenj) [reply frontend/src/components/application-form.js:5]: Labels should surround their inputs or have an `htmlFor` attribute.

File: frontend/src/components/application-form.js [hunk 3]
```diff
@@ -1,12 +1,59 @@
...
-    field_name: the key name for the field
-    validate: possible validations, not yet implemented
-    required: whether or not the field is required to be filled
-*/
+import React from "react";
+
+export const DefaultInput = ({ label, curValue, onChange, required }) => (
+    <div>
+        <label style={{ width: "20vw" }}>
+            {required ? label + ":" : label}
```
Comment (siefkenj) [inline frontend/src/components/application-form.js:6]: A `:` indicates a required field? That's a bit subtle...

File: frontend/src/views/application_form/actions.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { UPDATE_FIELD } from "./constants";
 import { error, success } from "react-notification-system-redux";
-import { errorProps } from "../notifications/constants";
+import { errorProps } from "../../modules/notifications/constants";
 import { apiPOST } from "../../libs/apiUtils";
 
 // an action generator function that returns an action object
```
Comment (siefkenj) [inline frontend/src/views/application_form/actions.js:3]: actions defined in a view should only relate to UI state, not model state. All actions for changing/updating model state and interacting with the API are in `/api`. There is no reason a view should ever import `apiPOST`, for example.

File: frontend/src/components/application-form.js (hunk 1 above)
Comment (thezane) [reply frontend/src/components/application-form.js:5]: The application form will be restyled and redesigned in a new issue.  Some of this code may be replaced.  I recommend we make changes in the new issue.

File: frontend/src/components/application-form.js (hunk 2 above)
Comment (thezane) [reply frontend/src/components/application-form.js:49]: Same as first comment

File: frontend/src/components/application-form.js (hunk 3 above)
Comment (thezane) [reply frontend/src/components/application-form.js:6]: Same as first comment

File: frontend/src/views/application_form/actions.js (hunk 4 above)
Comment (thezane) [reply frontend/src/views/application_form/actions.js:3]: I refactored part of the actions into api/actions.  Let me know of what you think.

File: frontend/src/api/actions/sessions.js [hunk 5]
```diff
@@ -94,6 +94,8 @@ export const setActiveSession = validatedApiDispatcher({
         // Save their return values so that we can await them after they're
         // dispatched. This way the browser can do parallel fetching.
         const promises = [];
+
+        //eslint-disable-next-line
```
Comment (siefkenj) [inline frontend/src/api/actions/sessions.js]: This must be an eslint issue. It should be fixed in a different PR.

File: frontend/src/api/actions/sessions.js [hunk 6]
```diff
@@ -94,6 +94,7 @@ export const setActiveSession = validatedApiDispatcher({
         // Save their return values so that we can await them after they're
         // dispatched. This way the browser can do parallel fetching.
         const promises = [];
+
```
Comment (siefkenj) [inline frontend/src/api/actions/sessions.js]: Remove extra line

Comment (thezane) [reply frontend/src/api/actions/sessions.js]: Removed
